### DISCUSSION TITLE
Automatic downloading of beatmaps in multiplayer

### DISCUSSION THREAD

**Discussion by lexisother** [ppy/osu#28431] [Ideas]
Currently when enabling the "Automatically download missing beatmaps", judging by the code that uses the option it only applies to spectating and watching replays. Could be nice to also have this apply to beatmaps that users add in multiplayer lobbies.

**Comment by Stoppedpuma** [comment]
Considering the response of https://github.com/ppy/osu/discussions/13204#discussioncomment-1549979, probably not since there can be huge queues in case of large lobbies?

**Reply by M1N0RM1N3R** [reply]
Maybe instead of downloading the entire queue in one click, it could be like a toggle (or even a slider) on the lobby screen and/or in the settings to automatically download the next (few) map(s) in the queue when you join/return to the lobby?
